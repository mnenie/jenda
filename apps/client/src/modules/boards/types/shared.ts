import type { User } from "@/modules/auth/types"

type S<T extends string> = T

export interface Label {
  id?: string
  name: string
  color: string
  taskCount?: number
  status?: S<'default' | 'editing'>
}

export type Status = S<'active' | 'archived'>

export type Visibility = 'private' | 'shared'

export type Priority = 'none' | 'low' | 'medium' | 'high'


export interface LabelWithStatus extends Label {
  status: S<'default' | 'editing'>
}

export type UserOption = Omit<User, 'role'>
