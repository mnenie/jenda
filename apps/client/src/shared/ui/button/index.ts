import { cva, type VariantProps } from 'class-variance-authority'

export { default as UiButton } from './UiButton.vue'

export const buttonVariants = cva('btn', {
  variants: {
    variant: {
      default: 'btn-default',
      solid: 'btn-solid',
      destructive: 'btn-colorized-red',
      success: 'btn-colorized-green',
      outline: 'btn-outline',
      secondary: 'btn-secondary',
      ghost: 'btn-ghost',
      dashed: 'btn-dashed',
    },
    size: {
      default: 'h-9 2xl:h-8 rounded-md px-2',
      xs: 'h-8 rounded px-2 text-xs rounded-lg',
      md: 'h-9.5 2xl:h-9 rounded-md px-2.5 py-1.5',
      sm: 'h-9 2xl:h-8 rounded-md px-3',
      lg: 'h-9 sm:h-10 2xl:h-9 px-4 py-2',
    },
  },
})

export type ButtonVariants = VariantProps<typeof buttonVariants>
