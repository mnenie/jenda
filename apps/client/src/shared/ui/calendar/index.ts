export { default as UiCalendar } from './UiCalendar.vue'
