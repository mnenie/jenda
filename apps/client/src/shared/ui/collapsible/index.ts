export { default as UiCollapsible } from './UiCollapsible.vue'
export { default as UiCollapsibleContent } from './UiCollapsibleContent.vue'
export { default as UiCollapsibleTrigger } from './UiCollapsibleTrigger.vue'
