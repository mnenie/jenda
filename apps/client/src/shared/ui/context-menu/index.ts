export { default as UiContextMenu } from './UiContextMenu.vue'
export { default as UiContextMenuContent } from './UiContextMenuContent.vue'
export { default as UiContextMenuGroup } from './UiContextMenuGroup.vue'
export { default as UiContextMenuItem } from './UiContextMenuItem.vue'
export { default as UiContextMenuLabel } from './UiContextMenuLabel.vue'
export { default as UiContextMenuSeparator } from './UiContextMenuSeparator.vue'
export { default as UiContextMenuShortcut } from './UiContextMenuShortcut.vue'
export { default as UiContextMenuTrigger } from './UiContextMenuTrigger.vue'
