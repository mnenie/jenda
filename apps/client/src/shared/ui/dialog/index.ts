export { default as UiDialog } from './UiDialog.vue'
export { default as UiDialogClose } from './UiDialogClose.vue'
export { default as UiDialogContent } from './UiDialogContent.vue'
export { default as UiDialogDescription } from './UIDialogDescription.vue'
export { default as UiDialogFooter } from './UiDialogFooter.vue'
export { default as UiDialogHeader } from './UiDialogHeader.vue'
export { default as UiDialogScrollContent } from './UiDialogScrollContent.vue'
export { default as UiDialogTitle } from './UiDialogTitle.vue'
export { default as UiDialogTrigger } from './UiDialogTrigger.vue'
