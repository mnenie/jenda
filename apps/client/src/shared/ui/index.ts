export * from './alert'
export * from './badge'
export * from './button'
export * from './calendar'
export * from './card'
export * from './checkbox'
export * from './collapsible'
export * from './command'
export * from './context-menu'
export * from './data-table'
export * from './dialog'
export * from './dropdown-menu'
export * from './dropdown-menu'
export * from './form'
export * from './input'
export * from './pagination'
export * from './picker'
export * from './pin-input'
export * from './popover'
export * from './select'
export * from './sheet'
export * from './shimmer-button'
export * from './switch'
export * from './table'
export * from './tabs'
export * from './tags-input'
export * from './textarea'
