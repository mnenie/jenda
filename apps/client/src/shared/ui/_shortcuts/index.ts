import { alert } from './alert'
import { badge } from './badge'
import { btn } from './button'
import { dialog } from './dialog'
import { dropdownMenu } from './dropdown-menu'
import { form } from './form'
import { input } from './input'
import { pinInput } from './pin-input'
import { select } from './select'
import { popover } from './popover'
import { command } from './command'
import { tabs } from './tabs'
import { pagination } from './pagination'
import { table } from './table'
import { checkbox } from './checkbox'
import { picker } from './picker'
import { tagsInput } from './tags-input'
import { _switch } from './switch'
import { textarea } from './textarea'
import type { Preset, StaticShortcutMap } from '@unocss/core'

export const shortcuts = [
  ...btn,
  ...input,
  ...textarea,
  ...badge,
  ...alert,
  ...dropdownMenu,
  ...select,
  ...dialog,
  ...pinInput,
  ...form,
  ...popover,
  ..._switch,
  ...command,
  ...tabs,
  ...pagination,
  ...checkbox,
  ...table,
  ...picker,
  ...tagsInput,
] as Exclude<Preset['shortcuts'], undefined | StaticShortcutMap>
