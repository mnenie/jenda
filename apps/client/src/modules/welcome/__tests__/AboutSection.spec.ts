import { shallowMount } from '@vue/test-utils'
import { describe, expect, it, vi } from 'vitest'
import AboutSection from '../components/sections/AboutSection.vue'
import { UiButton } from '@/shared/ui'

const mockRouter = {
  push: vi.fn(),
  beforeEach: vi.fn(),
}

describe('tests for AboutSection.vue', () => {
  const wrapper = shallowMount(AboutSection, {
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  })

  it('should be render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should redirect correctly', async () => {
    const btn = wrapper.find('.btns').findAllComponents(UiButton).at(0)
    await btn.trigger('click')
    expect(mockRouter.push).toHaveBeenCalledWith({ name: 'sign-up' })
  })
})
