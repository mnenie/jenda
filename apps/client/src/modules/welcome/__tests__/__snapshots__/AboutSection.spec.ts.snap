// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for AboutSection.vue > should be render correctly 1`] = `
"<section id="about" class="flex flex-col items-center py-0 px-1 gap-1 my-0 mx-auto max-w-980px max-[520px]:gap-2">
  <div class="bg-neutral-100 text-neutral-500 flex items-center gap-1 px-2.5 py-0.5 rounded-lg cursor-pointer dark:bg-neutral-700 dark:text-neutral-400 max-[520px]:mb-4 max-[520px]:justify-center"><span style="margin-right: 3px;">âœ¨</span><span class="text-default">Core: Task Boards, Real-Time Tracking</span><span i-hugeicons-arrow-right-01="" class="ease-in duration-100" style="margin-left: 0px;"></span></div>
  <h1 class="text-76px leading-tight font-semibold mb-1 text-center tracking-tighter max-[1220px]:text-68px max-[1152px]:text-64px max-[1100px]:text-62px max-[890px]:text-54px max-[680px]:!text-44px">Your path to perfection</h1>
  <p class="text-lg max-w-600px text-center text-neutral-500 dark:text-neutral-400 max-[680px]:max-w-480px max-[680px]:text-base">Jenda is a cloud-based program for efficient collaborative and individual project and task management.</p>
  <div class="flex justify-center items-center gap-4 px-0 py-4 btns">
    <ui-button-stub variant="default" size="lg" loading="false" disabled="false" loadingplacement="leading" aschild="false" as="button"></ui-button-stub>
    <ui-button-stub variant="outline" size="lg" loading="false" disabled="false" loadingplacement="leading" aschild="false" as="button"></ui-button-stub>
  </div>
</section>"
`;
